src/main/scala/mesosphere/marathon/api/serialization/ContainerSerializer.scala

```diff
@@ -2,49 +2,49 @@ package mesosphere.marathon.api.serialization
 
 import mesosphere.marathon.Protos
 import mesosphere.marathon.core.externalvolume.ExternalVolumes
-import mesosphere.marathon.state.Container.Docker.PortMapping
+import mesosphere.marathon.state.Container.DockerDocker.PortMapping
 import mesosphere.marathon.state._
 import org.apache.mesos
 
 import scala.collection.JavaConverters._
 import scala.collection.immutable.Seq
 
 object ContainerSerializer {
+  def fromProto(proto: Protos.ExtendedContainerInfo): Container = {
```

Comment:
The specific `fromProto` to which we delegate from here do not check whether the respective proto field is set. Instead, they directly access `getDocker`, `getMesosDocker`, `getMesosAppC`, which will lead to runtime exceptions when the caller of these specific `fromProto` functions does not correctly verify whether the fields are set, or when the functions are called from another location.

I'd strongly suggest to make this type- and nullsafe. One solution would be to provide partial functions that only match if the given field is set